flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/fakes/StorageClientFaker.java

```diff
@@ -580,57 +580,70 @@ public static BigQueryReadOptions createReadOptions(
                 .build();
     }
 
-    public static BigQueryReadOptions createTableReadOptions(
+    public static BigQueryReadOptions createReadAndWriteOptions(
             Integer expectedRowCount,
             Integer expectedReadStreamCount,
             String avroSchemaString,
-            Integer readLimit)
-            throws IOException {
-        return createTableReadOptions(
-                expectedRowCount,
-                expectedReadStreamCount,
-                avroSchemaString,
-                params -> StorageClientFaker.createRecordList(params),
-                0D,
-                readLimit);
-    }
-
-    public static BigQueryReadOptions createTableReadOptions(
-            Integer expectedRowCount, Integer expectedReadStreamCount, String avroSchemaString)
+            SerializableFunction<RecordGenerationParams, List<GenericRecord>> dataGenerator,
+            AppendRowsResponse appendRowsResponse)
             throws IOException {
-        return createTableReadOptions(
+        return createReadAndWriteOptions(
                 expectedRowCount,
                 expectedReadStreamCount,
                 avroSchemaString,
-                params -> StorageClientFaker.createRecordList(params),
+                dataGenerator,
+                appendRowsResponse,
                 0D,
                 -1);
     }
 
-    public static BigQueryReadOptions createTableReadOptions(
+    public static BigQueryReadOptions createReadAndWriteOptions(
```

Comment:
What is the reason for changing the method to create a read and write options? Btw, it is still returning a BiqQueryReadOptions object. Something does not seem right here.

Reply:
I think the diff is causing confusion. 
Added a new method `createReadAndWriteOptions()` which initializes `FakeBigQueryServices` instance. `FakeBigQueryServices` has two params `FakeBigQueryStorageReadClient` and `FakeBigQueryStorageWriteClient`. `createTableReadOptions ()` sets `FakeBigQueryStorageWriteClient` to null as the read test does not require a `FakeBigQueryStorageWriteClient`. `createTableReadOptions()` were initially 4 methods with different signatures(depending upon the params provided, reduced them to 2 (the actually required method)) and added `createReadAndWriteOptions()` method for instantiating both `FakeBigQueryStorageReadClient` and `FakeBigQueryStorageWriteClient`